griptape/drivers/vector/astra_db_vector_store_driver.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import logging
+import warnings
+from typing import TYPE_CHECKING, Any, Optional
+
+from attrs import define, field
+
+from griptape.drivers import BaseVectorStoreDriver
+from griptape.utils import import_optional_dependency
+
+if TYPE_CHECKING:
+    from astrapy import Collection
+    from astrapy.authentication import TokenProvider
+
+GRIPTAPE_VERSION: Optional[str]
+try:
+    from importlib import metadata
+
+    GRIPTAPE_VERSION = metadata.version("griptape")
+except Exception:
+    GRIPTAPE_VERSION = None
```

Comment:
This is interesting, though I feel like this should be done in a more generalized place rather than just this Vector Store Driver. Is this required for this PR?

Reply:
I can remove it, no problem.